### PR TITLE
add pytorch implementation and ut of autoencoder model for AE

### PR DESCRIPTION
**Pull request by cabuliwallah**
#4029

### CODE REVIEW

**Comment by TheaperDeng** [comment]
Possible reason for the different performance between tf and torch is the different default setting of learning rate:
https://pytorch.org/docs/stable/optim.html#torch.optim.Adadelta
https://www.tensorflow.org/versions/r1.15/api_docs/python/tf/keras/optimizers/Adadelta

File: pyzoo/zoo/chronos/model/anomaly/ae_detector.py [hunk 1]
```diff
@@ -90,10 +99,11 @@ def check_data(self, arr):
         if len(arr.shape) > 1:
             raise ValueError("Only univariate time series is supported")
 
-    def fit(self, y):
+    def fit(self, y, backend="keras"):
```
Comment (shane-huang) [inline pyzoo/zoo/chronos/model/anomaly/ae_detector.py:102]: move ```backend``` parameter to constructor. We want to keep fit interface as clean and light as possible

File: pyzoo/zoo/chronos/model/anomaly/ae_detector.py [hunk 2]
```diff
@@ -106,15 +116,39 @@ def fit(self, y):
...
+                         epochs=self.epochs,
+                         verbose=self.verbose)
+            y_pred = ae_model.predict(y)
+        else:
+            import torch.optim as optim
+            import torch.nn as nn
+            import torch
+            from torch.utils.data import TensorDataset, DataLoader
+            ae_model = create_torch_model(self.compress_rate, len(y[0]))
+            optimizer = optim.Adadelta(ae_model.parameters(), lr=0.001)
```
Comment (shane-huang) [inline pyzoo/zoo/chronos/model/anomaly/ae_detector.py:133]: add the ``lr``` parameter in the constructor as well.

**Comment by cabuliwallah** [comment]
Jenkins combination passed:
http://10.239.176.111:18888/job/ZOO-PR-Validation/5758/
http://10.239.176.111:18888/job/ZOO-PR-Validation-Spark-2.4/4135/
